test/verify/test_instancenorm.cpp

```diff
@@ -0,0 +1,97 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "verify_program.hpp"
+#include <migraphx/program.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/generate.hpp>
+#include <migraphx/common.hpp>
+#include <migraphx/make_op.hpp>
+
+migraphx::instruction_ref add_instancenorm(migraphx::module& m,
+                                           migraphx::instruction_ref x,
+                                           std::vector<size_t> dims,
+                                           float eps = 1e-5f)
+{
+    auto mgx_type = x->get_shape().type();
+    auto x_lens   = x->get_shape().lens();
+    std::vector<size_t> axes(x_lens.size() - 2);
+    std::iota(axes.begin(), axes.end(), 2);
+    auto scale   = m.add_parameter("scale", migraphx::shape{mgx_type, dims});
+    auto bias    = m.add_parameter("bias", migraphx::shape{mgx_type, dims});
+    auto epsilon = m.add_literal(migraphx::literal{migraphx::shape{mgx_type}, {eps}});
+
+    auto mean = m.add_instruction(migraphx::make_op("reduce_mean", {{"axes", axes}}), x);
+    auto mean_mbcast =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), mean);
+    auto sub = m.add_instruction(migraphx::make_op("sub"), x, mean_mbcast);
+    auto l0  = m.add_instruction(migraphx::make_op("sqdiff"), {x, mean_mbcast});
+    auto var = m.add_instruction(migraphx::make_op("reduce_mean", {{"axes", axes}}), {l0});
+    auto epsilon_mbcast =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), epsilon);
+    auto var_mbcast =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), var);
+    auto add_epsilon = m.add_instruction(migraphx::make_op("add"), var_mbcast, epsilon_mbcast);
+    auto rsqrt       = m.add_instruction(migraphx::make_op("rsqrt"), add_epsilon);
+    auto l1          = m.add_instruction(migraphx::make_op("mul"), {sub, rsqrt});
+    auto scale_mbcast =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), scale);
+    auto mul = m.add_instruction(migraphx::make_op("mul"), scale_mbcast, l1);
+    auto bias_mbcast =
+        m.add_instruction(migraphx::make_op("multibroadcast", {{"out_lens", x_lens}}), bias);
+    return m.add_instruction(migraphx::make_op("add"), mul, bias_mbcast);
+}
+
+template <migraphx::shape::type_t TYPE>
+struct test_instancenorm : verify_program<test_instancenorm<TYPE>>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        auto* mm                 = p.get_main_module();
+        std::vector<size_t> dims = {1, 2, 5, 5};
+        auto x                   = mm->add_parameter("x", migraphx::shape{TYPE, dims});
+        add_instancenorm(*mm, x, {1, 2, 1, 1});
+        return p;
+    }
+};
+template struct test_instancenorm<migraphx::shape::float_type>;
+template struct test_instancenorm<migraphx::shape::half_type>;
```

Comment:
half tests works without conversion to fp32 in verify for some reason. Therefore, didn't add converts for half.